### ISSUE TITLE
E721 regression(s)

### ISSUE THREAD

**Issue by asottile** [PyCQA/pycodestyle#1084]
so #1041 brought up a lot more instances of E721 and I don't feel confident adding this in a release since it's going to cause a bunch of (imo) unnecessary work

for example, here's something which wasn't flagged before but is now:

```python
assert type(x) is not bytes
```

strictly speaking the lint rule probably _should_ match this, but almost every case I can find with `is` and `is not` is being very explicit about the types being checked

I think the most reasonable thing to do here is to only check `==` and `!=` -- those are the ones that are most likely actual mistakes rather than an explicit check.

the other option is revert #1041 -- which might be the quickest thing to do to not block a release -- unclear what the best approach is here

**Comment by asottile** [comment]
I'm going to revert the change for now to unblock the release -- and we can reconsider that later given the above.  I think probably the most practical thing is to change the error message to suggest `isinstance(...)` or `is` / `is not` for exact comparisons and adjust the lint  rule to only error on `==` and `!=`.  but that's a lot for me to decide without outside input and reverting is uncontroversial for now

**Comment by asottile** [comment]
I put up my proposal above as #1086

**Comment by jepperaskdk** [comment]
Will this PR also fix this scenario?:
```
isinstance(False, int) # True
type(False) == int # False
```
I want to test if a value is exactly an int, not any type that inherits from it.

**Comment by asottile** [comment]
@jepperaskdk just try it
